### PR TITLE
[common] modify index type for PCLImage

### PR DESCRIPTION
**Pull request by haritha-j**

### CODE REVIEW

File: common/include/pcl/PCLImage.h [hunk 1]
```diff
@@ -5,19 +5,20 @@
...
 
-    std::uint32_t height = 0;
-    std::uint32_t width = 0;
+    index_t height = 0;
+    index_t width = 0;
     std::string encoding;
 
     std::uint8_t is_bigendian = 0;
-    std::uint32_t step = 0;
+    index_t step = 0;
```
Comment (SergioRAgostinho) [inline common/include/pcl/PCLImage.h:21]: Definite ABI break and potential API break.

File: common/include/pcl/PCLImage.h [hunk 2]
```diff
@@ -5,19 +5,20 @@
 #include <ostream>  // for ostream
 
 #include <pcl/PCLHeader.h>   // for PCLHeader
+#include <pcl/types.h> //for index_t
 
 namespace pcl
 {
   struct PCLImage
   {
      ::pcl::PCLHeader  header;
```
Comment (kunaltyagi) [inline common/include/pcl/PCLImage.h:14]: Should we reorder the memory layout to make it more cache friendly? We're dumping away at API and ABI stability anyways

Comment (SergioRAgostinho) [reply common/include/pcl/PCLImage.h:14]: Sure. What your proposal?

Comment (kunaltyagi) [reply common/include/pcl/PCLImage.h:14]: Maybe no change. As I started describing, I saw it'd cost too much (in time) and not provide comparable benefits

**Pull request merged**